pkg/catalog/fake.go

```diff
@@ -105,3 +106,108 @@ func NewFakeMeshCatalog(kubeClient kubernetes.Interface) *MeshCatalog {
 	return NewMeshCatalog(mockKubeController, kubeClient, meshSpec, certManager,
 		mockIngressMonitor, stop, cfg, endpointProviders...)
 }
+
+func newFakeMeshCatalog() *MeshCatalog {
```

Comment:
might be good to rename this func to something that says this is a fake mesh catalog for the ingress test

Reply:
Sure, maybe in a different change because it is not even used by ingress tests. Many tests are using this so it didn't make sense to change all tests in this change.

Reply:
yea sounds good. we keep wanting to get rid of this anyway so we can do that later.